### PR TITLE
Add store diff fact

### PR DESCRIPTION
**Pull request by lucperkins**
Another bit of info for PostHog to chew on.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -42,6 +46,9 @@ class MagicNixCacheAction extends DetSysAction {
     });
 
     this.hostAndPort = inputs.getString("listen");
+    this.diffStore = inputs.getBool("diff-store");
+
+    this.addFact(FACT_DIFF_STORE, this.diffStore);
```
Comment (grahamc) [inline src/index.ts:51]:
```suggestion
    this.addFact(FACT_DIFF_STORE_ENABLED, this.diffStore);
```

?

File: src/index.ts [hunk 2]
```diff
@@ -14,6 +14,10 @@ import { inspect } from "node:util";
 // twice.
 const ENV_DAEMON_DIR = "MAGIC_NIX_CACHE_DAEMONDIR";
 
+const FACT_AUTHENTICATED_ENV = "authenticated_env";
```
Comment (grahamc) [inline src/index.ts:17]: what does this mean exactly? Is there a more descriptive name?

Comment (lucperkins) [reply src/index.ts:17]: I'm not sure, all I've done here is `const`ify the key. Open to suggestion.

Comment (lucperkins) [reply src/index.ts]: I'm still not sure why we need this but I've settled on `required_env_vars_present` for now.

**Review by cole-h — APPROVED** [review]

**Pull request merged**